docs/extensions/testing-and-publishing/publishing.md

```diff
@@ -0,0 +1,47 @@
+# Publishing Extensions
+
+To be able to easily share extensions with users they need to be published somewhere.
+Lens only supports installing extensions from tarballs.
+All hosted extensions must, therefore, be retrievable in a tarball.
+
+## Places To Host Your Extension
+
+We recommend to host your extension somewhere on the web so that it is easy for people to search for and download it.
+We recommend either hosting it as an NPM package on https://www.npmjs.com or through GitHub releases.
+We recommend against using GitHub packages (https://github.com/features/packages) as it requires a GitHub token to access the package.
+
+### Publishing via NPM
+
+This is the easiest method of publishing as NPM comes built in with mechanism to get a link to download the package as a tarball.
+Once you have set up an account with NPM (https://www.npmjs.com/signup) and logged in with their CLI (`npm login`) you will be ready to publish.
+
+* Run `npm version <major|minor|patch>` to bump the version of your extension by the appropriate amount.
+* Run `npm publish` to publish your extension to NPM
+* RUn `git push && git push --tags` to push the commit that NPM creates to your git remote.
+
+It is probably a good idea to put into your README.md the following instructions for your users to get the tarball download link.
+
+```bash
+npm view <extension-name> dist.tarball
+```
+
+This will output the link that they will need to give to lens to install your extension.
+
+### Publish via GitHub Releases
+
+Another method of publishing your extensions is to do so with the releases mechanism built into GitHub.
+We recommend reading [GitHub's Releases Documentation](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/managing-releases-in-a-repository) for how to actually do the steps of a release.
+The following will be a quick walk through on how to make the tarball which will be the released file.
+
+### Making a Tarball of your extension
```

Comment:
I think `npm pack` is the easiest way to create a valid tarball?

Reply:
Ah cool, didn't know about that one. I was stumbling around trying to use `tar` and `find` and then I found this. I tried it on the lens repo and `npm pack` adds a bunch of not-needed files (ones that are `.gitignore`-ed). Should I add that as another option or keep this?

Reply:
Plain tarball is not a valid npm package so I think we should only document `npm pack`.

Reply:
I meant this as a way for creating the tarball for github releases but sure I can document `npm pack` instead.

Reply:
Plain tarball does not work, Lens accepts only npm packages .